templates/repo/diff/box.tmpl

```diff
@@ -124,25 +124,25 @@
 															<span rel="{{if $line.LeftIdx}}diff-{{Sha1 $file.Name}}L{{$line.LeftIdx}}{{end}}"></span>
 														</td>
 														<td class="lines-type-marker">
-															<pre>{{if $line.LeftIdx}}<span class="line-type-marker" data-type-marker="{{$line.GetLineTypeMarker}}"></span>{{end}}</pre>
+															{{if $line.LeftIdx}}<code data-type-marker="{{$line.GetLineTypeMarker}}"></code>{{end}}
 														</td>
 														<td class="lines-code lines-code-old halfwidth">
 															{{if and $.SignedUserID $line.CanComment $.PageIsPullFiles (not (eq .GetType 2))}}
 																<a class="ui green button add-code-comment add-code-comment-left" data-path="{{$file.Name}}" data-side="left" data-idx="{{$line.LeftIdx}}">+</a>
 															{{end}}
-															<pre><code class="wrap {{if $highlightClass}}language-{{$highlightClass}}{{else}}nohighlight{{end}}">{{if $line.LeftIdx}}{{$section.GetComputedInlineDiffFor $line}}{{end}}</code></pre>
+															<code class="wrap {{if $highlightClass}}language-{{$highlightClass}}{{else}}nohighlight{{end}}">{{if $line.LeftIdx}}{{$section.GetComputedInlineDiffFor $line}}{{end}}</code>
```

Comment:
So I suspect if you move the `{{if` to wrap the `<code>` the blank lines you're seeing might disappear.

Reply:
Hmm that doesn't make much sense.

Also noticed this is broken on master, so the breakage actually does not come from my changes I think:

https://try.gitea.io/silverwind/symlink-test/commit/f2f2a03c79d7b2ab89785fca78c2f85d123b7d78?style=split

Reply:
Another example: https://try.gitea.io/silverwind/symlink-test/commit/9a7f17dcf656d7c0d8bb0ec534c7746cfa349b3a?style=split

Unified diff is okay, but split diff is missing content on both sides. The right side notably has a absent `-` marker too. Maybe something in the template vars.

Reply:
Tried a few things, nothing seems to work. When I revert https://github.com/go-gitea/gitea/commit/5908bb103003912bd0ba3cefa66f2ff815ee8d8e, split diff works again so it certainly is something in that commit.

Reply:
That's weird as hell. What's the cause in 5908bb1?

Reply:
OK they're being rendered but they're being hidden by something.

Reply:
I genuinely don't understand this was definitely working before I committed. What is doing this deletion?

Reply:
There's some JS that messes with the HTML when clicking `.add-code-comment` but I have a hard time seeing how that could interfere.

Reply:
The line comment added by that JS is also somewhat broken. Try clicking "+" when hovering near the beginning of the line here:

https://try.gitea.io/silverwind/symlink-test/pulls/3/files?style=unified&whitespace=

Reply:
Finally found it.

Reply:
Hope you don't mind I've pushed a fix and some fixes for arc-theme-green.

So sorry that I caused this and missed this. I genuinely don't understand why this didn't appear in my testing.